### PR TITLE
#NP-44745 doi update with duplicates

### PR DESCRIPTION
**Pull request by StigNorland**
…uplicates

### CODE REVIEW

File: customer-commons/src/test/java/no/unit/nva/customer/model/SecretManagerDoiAgentDaoTest.java [hunk 1]
```diff
@@ -35,14 +35,34 @@ void shouldFailDeserialize() {
...
-    void shouldUpdate() {
-        var doiAgentDto = createSampleCustomerDao().toCustomerDto()
-                              .getDoiAgent().addPassword(randomString());
+    void shouldUpdateDoiAgent() {
+        var doiAgentDto =
+            createSampleCustomerDao()
+                .toCustomerDto()
+                .getDoiAgent()
+                .addPassword(randomString());
         var secretManagerDoiAgentDao = new SecretManagerDoiAgentDao(doiAgentDto);
```
Comment (ketilaa) [inline customer-commons/src/test/java/no/unit/nva/customer/model/SecretManagerDoiAgentDaoTest.java:44]: You create a SecretManagerDoiAgentDao from a CustomerDao. Then merge with the same CustomerDao instance. What is actually being tested here? Is it just for coverage?

**Review by ketilaa — CHANGES_REQUESTED** [review]
See comments.

File: customer-commons/src/test/java/no/unit/nva/customer/model/SecretManagerDoiAgentDaoTest.java (hunk 1 above)
Comment (StigNorland) [reply customer-commons/src/test/java/no/unit/nva/customer/model/SecretManagerDoiAgentDaoTest.java:44]: it is mostly for coverage.

I am merging two instances of  SecretManagerDoiAgentDao, one with an updated password.

Comment (StigNorland) [reply customer-commons/src/test/java/no/unit/nva/customer/model/SecretManagerDoiAgentDaoTest.java:43]: renamed tests and hopefully made it clearer.

**Review by ketilaa — APPROVED** [review]

**Pull request merged**